packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -32,7 +32,7 @@ export function proxyMiddleware(
   httpServer: http.Server | null,
   config: ResolvedConfig
 ): Connect.NextHandleFunction {
-  const options = config.server.proxy!
+  const options = (config.server.proxy || config.preview.proxy)!
```

Comment:
What if both `config.server.proxy` and `config.preview.proxy` are defined in preview command? I think we should always use `config.preview.proxy` in that case. Perhaps we need to refactor the `proxyMiddleware` function to accept `proxy` and `logger` instead of `config` only so the consumer can tell what proxy config to use.

```js
export function proxyMiddleware(
   httpServer: http.Server | null,
   proxy: Record<string, string | ProxyOptions>,
   logger: Logger
): Connect.NextHandleFunction {
```

Reply:
Agree, that will make code clear and make sense. Thanks for your advice!